### ISSUE TITLE
[typescript-estree] Provide separate context for accessor-declarations

### ISSUE THREAD

**Issue by manuth** [typescript-eslint/typescript-eslint#1885]
**Repro**
* Use any sort of plugin which allows settings for self-declared contexts.
For instance: [`eslint-plugin-jsdoc/require-param`](https://github.com/gajus/eslint-plugin-jsdoc/blob/a4023308851719285383fa36ea7c40b66db4c8c9/.README/rules/require-param.md)
* Try to apply this rule to all functions besides accessors

```JSON
{
  "rules": {
    "jsdoc/require-param": [
      "warn",
      {
        "exemptedBy": [],
        "contexts": [
          "ArrowFunctionExpression",
          "FunctionDeclaration",
          "FunctionExpression",
          "MethodDefinition",
          "TSAbstractMethodDefinition"
        ]
      }
    ],
  }
}
```

```ts
/**
 * Test class
 */
export class Test
{
    /**
     * A test-variable.
     */
    private test = 1;

    /**
     * Gets or sets the test-variable.
     */
    public get TestProperty(): number
    {
        return this.value;
    }

    // I want this rule to not report a failure here (because jsdoc for `value` is missing)
    public set TestProperty(value)
    {
        this.test = value;
    }

    // While I want to have a failure here because in this case `value` is a meaningful parameter.
    public TestMethod(value): void
    {
        console.log(value);
    }
}
```

**Expected Result**
`typescript-estree` should provide two different node-types for method-definitions and get/set accessors.

**Actual Result**
Method-definitions and get/set-accessors have the same node-type and are therefore treated the same way by all rules.

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `2.27.0` |
| `TypeScript`                           | `3.8.3` |
| `node`                                 | `12.16.1` |
| `npm`                                  | `6.13.4` |

**Comment by bradzacher** [comment]
Whilst the rule calls the option "context", they aren't contexts - they're just the names of AST nodes.
The AST structure is defined not by us, but by a committee of sorts in ESTree: https://github.com/estree/estree/

Here is the AST of your provided code: https://astexplorer.net/#/gist/3d2c7ab68328d93413678f3ae4fe447f/364df6cee6d363cb87742053cf19ca7247716ad1

So there's nothing we can do here - the rule you linked must add support for it in some way.
